verisignmdns/src/main/java/denominator/verisignmdns/VrsnMdnsContentHandler.java

```diff
@@ -0,0 +1,167 @@
+                domainElementFound = true;
+            }
+        }
+
+        @Override
+        public void characters(char ch[], int start, int length) throws SAXException {
+            if (domainElementFound) {
+                zones.add(Zone.create(new String(ch, start, length)));
+            }
+        }
+
+        @Override
+        public void endElement(String uri, String localName, String qName) throws SAXException {
+            if (qName != null && qName.endsWith(":domainName")) {
+                domainElementFound = false;
+            }
+        }
+    }
+
+    static class RecordListHandler extends DefaultHandler implements ContentHandlerWithResult<List<Record>> {
+        private final List<Record> rrs = new ArrayList<Record>();
+
+        @Inject
+        RecordListHandler() {
+        }
+
+        private Record rr = new Record();
+        private boolean processingRR = false; // flag indicating currently
+                                              // inside resource record
+                                              // element..
+        private String currentElementName = "";
+        private String rDataString = "";
+
+        @Override
+        public List<Record> result() {
+            return rrs;
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes attrs) {
+
+            if (qName.endsWith(":resourceRecord")) {
+                rr = new Record();
+                processingRR = true;
+                currentElementName = ":resourceRecord";
+            }
+            if (processingRR && qName.endsWith(":resourceRecordId")) {
+                currentElementName = ":resourceRecordId";
+            }
+            if (processingRR && qName.endsWith(":owner")) {
+                currentElementName = ":owner";
+            }
+            if (processingRR && qName.endsWith(":type")) {
+                currentElementName = ":type";
+            }
+            if (processingRR && qName.endsWith(":ttl")) {
+                currentElementName = ":ttl";
+            }
+            if (processingRR && qName.endsWith(":rData")) {
+                currentElementName = ":rData";
+            }
+        }
+
+        @Override
+        public void endElement(String uri, String name, String qName) {
+
+            if (qName.endsWith(":resourceRecord")) {
+                rrs.add(rr);
+                processingRR = false;
+            }
+
+            if (qName.endsWith(":rData")) {
+                String[] tempArray = rDataString.split(",");
+                rr.rdata = getArrayList(tempArray);
+                rDataString = "";
+            }
+            currentElementName = "";
+        }
+
+        /**
+         * This method is to ensure all space characters are accounted for while
+         * processing rData
+         * 
+         */
+        @Override
+        public void ignorableWhitespace(char[] ch, int start, int length) throws SAXException {
```

Comment:
Please rewrite this to the same style as the other sax parsers. The intent of denominator is that code is similar. There's no special case for verisign sax parsers to be written with different callbacks than the others. If there is, make sure there's a test that clearly shows why. If there's something wrong with the style of sax used for the others, raise it up.